ckanext/qgov/common/plugin.py

```diff
@@ -636,3 +637,15 @@ def get_validators(self):
             'valid_url': valid_url,
             'valid_resource_url': valid_resource_url
         }
+
+    # IResourceController
+
+    def before_create(self, context, data_dict):
```

Comment:
should we look at breaking this into its own plugin, this plugin.py is getting very large on implements details.

Reply:
It was actually supposed to be doing something like this already, but our monkey-patch stopped working when we installed ckanext-s3filestore. I suppose I could remove the obsolete code.